packages/cmrjs/src/cmr-utils.js

```diff
@@ -1221,6 +1284,81 @@ async function getGranuleTemporalInfo(granule) {
   return {};
 }
 
+/**
+ * Search recursively for the location of a sub-object in the cmr object
+ *
+ * @param {object} cmrObject - cmr metadata object
+ * @param {string} attributePath - subObject path to seek
+ * @returns {string | null}
+*/
+const findCollectionAttributePath = (cmrObject, attributePath) => {
+  if (get(cmrObject, attributePath)) {
+    return attributePath;
+  }
+  let output = null;
+  Object.entries(cmrObject).forEach(([key, value]) => {
+    if (isObject(value)) {
+      const foundPath = findCollectionAttributePath(value, attributePath);
+      if (foundPath !== null) {
+        output = key + '.' + foundPath;
+      }
+    }
+  });
+  return output;
+};
+
+/**
+ *
+ * @param {object} cmrObject - CMR metadata object
+ * @param {string} identifierPath - path to identify this field by
+ * @param {string} value - value to set this identifier to
+ * @param {string | null} defaultPath - where to put value if identifier can't be found
+ */
+const updateCMRCollectionValue = (
+  cmrObject,
+  identifierPath,
+  value,
+  defaultPath = null
+) => {
+  const backupPath = defaultPath || identifierPath;
+  const fullPath = findCollectionAttributePath(cmrObject, identifierPath) || backupPath;
+  set(cmrObject, fullPath, value);
```

Comment:
We should probably avoid mutation here, or consider not abstracting this?   

I'm less concerned about the implementation as stands, but that this isn't by convention/etc something that someone would think twice about importing elsewhere at some point in the future.

Reply:
OBE